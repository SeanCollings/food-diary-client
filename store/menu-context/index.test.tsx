import { render } from '@testing-library/react';
import { MenuContextProvider, initialState, useMenuContext } from '.';
import { useEffect, useRef } from 'react';

const TestComponent = () => <div>Test component</div>;

const MockToggleMenuComponent: React.FC<{ toggleTotal: number }> = ({
  toggleTotal,
}) => {
  const toggleRef = useRef(1);
  const { isOpen, toggleMenu } = useMenuContext();

  useEffect(() => {
    if (toggleRef.current <= toggleTotal) {
      toggleMenu();
      toggleRef.current += 1;
    }
  }, [isOpen, toggleMenu, toggleTotal]);

  return <div>Open: {isOpen.toString()}</div>;
};

describe('store - menu-context', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return initial state', () => {
    const { isOpen, toggleMenu } = initialState;

    expect(isOpen).toBeFalsy();
    expect(toggleMenu()).toBeNull();
  });

  it('should render provider with children', () => {
    const { asFragment } = render(
      <MenuContextProvider>
        <TestComponent />
      </MenuContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should toggle menu open', () => {
    const { asFragment } = render(
      <MenuContextProvider>
        <MockToggleMenuComponent toggleTotal={1} />
      </MenuContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should toggle menu closed after being opened', () => {
    const { asFragment } = render(
      <MenuContextProvider>
        <MockToggleMenuComponent toggleTotal={2} />
      </MenuContextProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
