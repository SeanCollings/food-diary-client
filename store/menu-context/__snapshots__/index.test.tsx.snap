// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store - menu-context should render provider with children 1`] = `
<DocumentFragment>
  <div>
    Test component
  </div>
</DocumentFragment>
`;

exports[`store - menu-context should toggle menu closed after being opened 1`] = `
<DocumentFragment>
  <div>
    Open: false
  </div>
</DocumentFragment>
`;

exports[`store - menu-context should toggle menu open 1`] = `
<DocumentFragment>
  <div>
    Open: true
  </div>
</DocumentFragment>
`;
